app/serializers/api/v3/get_contexts/recolor_by_attribute_serializer.rb

```diff
@@ -0,0 +1,26 @@
+module Api
+  module V3
+    module GetContexts
+      class MRecolorByAttributeSerializer < ActiveModel::Serializer
+        attributes :is_default, :is_disabled, :group_number, :position, :legend_type, :legend_color_theme, :interval_count, :min_value, :max_value, :divisor,
+          :is_default, :is_disabled, :name
+        attribute :display_name, key: :label
+        attribute :tooltip_text, key: :description
+
+        attribute :type do
+          object.original_type.downcase
+        end
+
+        attribute :nodes do
+          flows = []
+          if (object.legend_type.eql? 'qual')
+            FlowQual.distinct.select('value').where('qual_id = ? AND value NOT ILIKE \'UNKNOWN%\'', object.original_id).each do |flow|
```

Comment:
this looks dodgy, as in this whole pushing in the loop could be replaced by a pluck no? the naming is also quite confusing to me; what this loop produces is an array of flow_qual values and it's in a variable called `flows`, returned to the client as `nodes`; at this point we don't want to change the response, but it would be great to document what this does and maybe we could revisit it when we're done with stabilising.

Reply:
dodgy is my code's middle name ;) Just to mention that this is not new code, although git sees it that way. But I'm sure there are 1001 improvement points all over the place, this is some of my very first rails code ;)

Reply:
Sure, I appreciate that is the case :) I think when there are small readability improvements to be made we should be making them as we migrate the code, and when there are bigger parts of hairy code we could flag them with # TODO: refactor or something to make sure we don't forget about them when we have stabilised the thing. This is a great opportunity to review our whole codebase, even if some things will need to wait to be fixed.

Reply:
We actually don't need this here, as this is controller-related logic. moved to the other PR